sdk/communication/AzureCommunicationChat/Source/Models/MessageContent.swift

```diff
@@ -47,25 +48,36 @@ public struct MessageContent: Codable {
     ///   - chatMessageContent: ChatMessageContent to initialize from.
     public init(
         from chatMessageContent: ChatMessageContent
-    ) {
+    ) throws {
         self.message = chatMessageContent.message
         self.topic = chatMessageContent.topic
-        self.participants = (chatMessageContent.participants != nil) ?
-            chatMessageContent.participants!.map { Participant(from: $0) } : nil
-        self.initiator = chatMessageContent.initiator
+
+        // Convert ChatParticipants to Participants
+        if let participants = chatMessageContent.participants {
+            self.participants = try participants.map { try Participant(from: $0) }
+        } else {
+            self.participants = nil
+        }
+
+        // Deserialize the identifier model to CommunicationIdentifier
+        if let identifierModel = chatMessageContent.initiatorCommunicationIdentifier {
+            self.initiator = try IdentifierSerializer.deserialize(identifier: identifierModel)
+        } else {
+            self.initiator = nil
+        }
```

Comment:
It seems like a simple model type should not throw.

Reply:
The IdentifierSerializer is what could throw @JoshuaLai not sure if this can be avoided

I did not want to just swallow the error so any method where the convenience layer has to convert types using the serializer now throws

Reply:
It's not convenient if the convenience layer forces the user to wrap calls in `do/catch`  that otherwise would not have. Since `nil` is an acceptable value for `initiator`, it would be okay to use `try?` to make them fail as nil, or perhaps you can catch the deserialization error here, log it, and still set `initiator` to `nil`. Then it doesn't silently fail since the issue will be in the logs, but the developer doesn't need more error handling.

Reply:
For Message and MessageContent where the identifier can be `nil` that sounds good. 

- `Add participant`, `remove participant` and `create thread` all use the serializer to convert the participants
- If the serializer fails it means we can't make the request to add/remove the participants
- In the case of creating a thread we could still technically make the request without the participants - but it might be confusing to the user as the response they receive would show that there were no errors adding participants

What about returning a failure to the completion handler? Error message could clarify that the request was never actually sent:
```
do {
  // Use the identifier serializer that throws in here
} catch {
    let azureError = AzureError.client("Failed to send create thread request.", error)
    completionHandler(.failure(azureError), nil)
}
```

Reply:
For `ChatClient` `create(thread)` and `ChatThreadClient``add(participant)`, `remove(participant` I've opted to return the failure to the user to handle instead of throwing, let me know if you have other thoughts.

For the initializers that now throw - `Participant`, `Message`, `ReadReceipt`, `Thread`, `MessageContent`, `CreateThreadResult` I've opted to keep them as throwing. Reasoning is that those initializers are used by the convenience layer but users of the SDK can use the non-throwing initializers.